query-engine/connectors/sql-query-connector/src/database/operations/write.rs

```diff
@@ -322,18 +321,52 @@ pub async fn update_records(
         write::update_many(model, ids.as_slice(), args, filter_condition, trace_id)?
     };
 
+    for update in updates {
+        conn.execute(update).await?;
+    }
+
+    Ok(merge_write_args(ids, id_args))
+}
```

Comment:
+1 on splitting the method in two instead of an enum parameter

Reply:
I feel this whole section could be completely cleaned up. But I didn't want to do that in this PR.